content/10-managing-databases/07-introduction-database-caching.mdx

```diff
@@ -0,0 +1,97 @@
+
+Data retrieval speed greatly affects the user experience of an application. Implementing a caching strategy on your database can result in improved database performance, availability, and scalability for minimal cost depending on the strategy, all factors contributing to an overall positive application experience.
+
+### Performance
+As touched on briefly, database caching improves the performance of a database by making data more easily accessed. The cache acts has a sort of “keyboard short-cut” or “hot-key” for the application to reference data that it frequently is calling upon. 
+
+This speedier request can minimize the workload of the database, keeping it from spending inefficient amounts of time doing repetitive tasks. Instead making these tasks more efficient and simplifying data access. 
+
+### Availability
+While not a 100% failover strategy, caching also provides benefits to overall database availability. Depending on where the cache is stored, the cache can still provide a place for the application to call upon for data in the case of the primary database server becoming unavailable for any reason.
+
+While database performance is generally the primary reason for adopting a caching strategy, you also have the added benefit of some additional resiliency in the case of any backend failures. 
+
+### Scalability
+Similarly to added high availability, database caching has a positive effect on scalability. While it shouldn’t be your main consideration for a database scaling strategy, implementing caching to improve database performance reduces your database workload, therefore distributing backend queries across entities. 
+
+This distribution lightens the load on a primary database and can reduce costs and provide more flexibility in the processing of your data. This results in a more scalable environment as you continue for take more frequently requested data off the plate of the primary database.
+
+## What are the different database caching strategies?
+
+Before adopting database caching into your data access flow, it is important to consider which caching strategy is best suited for the job. For any scenario, the relationship between the database and the cache can have a different impact on performance and system structure. Planning ahead and considering all options will lead to less headaches down the road. 
+
+The four most popular strategies to consider are cache aside, read through, write through, and write back. We’ll cover the data source to cache relationship and process of each strategy.
+
+### Cache-aside
+In a cache-aside arrangement, the database cache sits next to the database. When the application requests data, it will check the cache first. If the cache has the data (a cache hit), then it will return it. If the cache does not have the data (a cache miss), then the application will query the database. The database then stores that data in the cache for any subsequent application queries.
```

Comment:
One thing not clear to me here is which component is directing this process.  Is the logic to query the cache and then fallback to the main database built into the application's DB access code?  Or is it something that the cache itself handles (application only knows to query the cache. The cache returns results if it has them, and if not it fetches them from the DB and updates its store for next time).  I don't think you need to say a lot about it, but clarifying who the "actor" is in these sentences might be helpful.  (For instance, I don't know if most databases have the logic to reach out and update a cache by themselves in the last sentence).

Reply:
Okay, so after looking at the next one, it looks like maybe the application is in control of this request flow here.  I think that I understand a bit better, but it still feels uncertain about what triggers that last action.  [This](https://docs.microsoft.com/en-us/azure/architecture/patterns/_images/cache-aside-diagram.png) implies that it's the application that's explicitly updating the cache.

Reply:
I believe you are correct and this is my mistake. It should be the application.